tests/test_sync.py

```diff
@@ -92,56 +87,54 @@ def test_local_equals_API(self, pir_to_xy_response, downsample_ref):
         assert y_pred == pytest.approx(y, abs=1e-2)
 
 
-class TestDownloadParallelDataset:
-    @pytest.mark.parametrize("include_expression", [True, False])
-    @pytest.mark.parametrize("downsample_ref", [25, 50])
-    @pytest.mark.parametrize("axis", ["coronal", "sagittal"])
-    def test_patched(self, include_expression, downsample_ref, axis, monkeypatch):
-        """Does not requires internet, everything is patched.
-
-        The only thing that is unpatched is the `get_transform_simple`.
-        """
+class TestDatasetDownloader:
+    @responses.activate
+    def test_patched(self, data_folder, custom_cache_dir):
+        """Does not requires internet, everything is patched."""
 
         # Parameters
-        dataset_id = 12345
-
-        # Mocking
-        get_2d_bulk_fake = Mock(
-            return_value={
-                11111: (np.ones((2, 3)), 20),
-                22222: (np.ones((2, 3)), 50),
-            }
+        dataset_id = 123
+        include_expression = False
+        downsample_ref = 25
+        axis = "sagittal"
+
+        # Mocking the responses to request
+        with open(data_folder / "sync" / "SectionDataSet.txt") as f:
+            body_dataset = f.read()
+
+        responses.add(
+            responses.GET,
+            "https://api.brain-map.org/api/v2/data/query.json?"
+            "criteria=model::SectionDataSet,rma::criteria,[id$eq123],"
+            "rma::include,alignment3d,rma::options[start_row$eq0][num_rows$eq25000]",
+            body=body_dataset,
         )
 
-        get_3d_fake = Mock(
-            return_value=np.ones((3, 4)),
-        )
+        with open(data_folder / "sync" / "SectionImage.txt") as f:
+            body_image = f.read()
 
-        common_queries_fake = Mock()
-        common_queries_fake.get_axis.return_value = axis
-
-        get_image_fake = Mock(
-            return_value=np.zeros((10, 10)),
+        responses.add(
+            responses.GET,
+            "https://api.brain-map.org/api/v2/data/query.json?"
+            "criteria=model::SectionImage,rma::criteria,"
+            "%5Bdata_set_id$eq123%5D,rma::include,alignment2d,"
+            "rma::options%5Bstart_row$eq0%5D%5Bnum_rows$eq25000%5D",
+            body=body_image,
         )
 
-        xy_to_pir_fake = Mock(return_value=(1200, 235, 242))
-
-        # Patching
-        monkeypatch.setattr("atldld.sync.get_2d_bulk", get_2d_bulk_fake)
-        monkeypatch.setattr("atldld.sync.get_3d", get_3d_fake)
-        monkeypatch.setattr("atldld.sync.get_image", get_image_fake)
-        monkeypatch.setattr("atldld.sync.CommonQueries", common_queries_fake)
-        monkeypatch.setattr("atldld.sync.xy_to_pir_API_single", xy_to_pir_fake)
+        # Save fake image in cache folder
+        custom_cache_dir.mkdir(exist_ok=True)
+        fake_img = Image.fromarray(np.zeros((100, 200), dtype=np.uint8))
+        fake_img.save(custom_cache_dir / "101945191-0.jpg")
```

Comment:
Similar to my comment above. I think it is better (+ simpler) if we patch `atldld.utils.get_image` directly. This is more in the spirit of unit testing, since in this test we are focusing on `DatasetDownloader` and we can make an assumption that "atldld.utils.get_image" works as expected.

Reply:
Patched `get_image` in cae39ef.